plugins/Actions/VisitorDetails.php

```diff
@@ -160,6 +162,14 @@ public function extendActionDetails(&$action, $nextAction, $visitorDetails)
             unset($action['url_prefix']);
         }
 
+        if (array_key_exists('url', $action) && strpos($action['url'], 'http://') === 0) {
+            $host = parse_url($action['url'], PHP_URL_HOST);
+
+            if ($host && PageUrl::shouldUseHttpsHost($visitorDetails['idSite'], $host)) {
+                $action['url'] = 'https://' . Common::mb_substr($action['url'], 7 /* = strlen('http://') */);
+            }
+        }
```

Comment:
Can we make this and the above change visible in API tests? Eg, setting the site URL of one host in an existing test to https so part of the output changes.

Reply:
will add/change some tests